packages/block-library/src/navigation/edit.js

```diff
@@ -76,6 +84,49 @@ function Navigation( {
 		[ pages ]
 	);
 
+	useEffect( () => {
+		// Indicate the fetching status
+		isRequestingPages = true;
+
+		const baseUrl = '/wp/v2/pages';
+
+		// "view" is required to ensure Pages are returned by REST API
+		// for users with lower capabilities such as "Contributor" otherwise
+		// Pages are not returned in the request if "edit" context is set
+		const context = 'view';
+
+		const filterDefaultPages = {
+			parent: 0,
+			order: 'asc',
+			orderby: 'id',
+			context,
+		};
+
+		apiFetch( {
+			path: addQueryArgs( baseUrl, filterDefaultPages ),
+		} ).then(
+			( pagesList ) => {
+				if ( isMounted ) {
+					setPages( pagesList );
+				}
+				// We've stopped fetching
+				isRequestingPages = false;
+			}
+		).catch(
+			() => {
+				if ( isMounted ) {
+					setPages( [] );
+				}
+				// We've stopped fetching
+				isRequestingPages = false;
+			}
+		);
+
+		return () => {
+			isMounted = false;
+		};
+	}, [] );
```

Comment:
I assume any component using apiFetch will hhave the same state, the same effects (status, data, error), should we extract this into a reusable `useApiFetch` hook like suggested?

Reply:
Yes we could should but I wanted to check the basics were on track first before commiting additional effort.

Ill take this as thumbs up to proceed with the hook 😁